### PR TITLE
telfhash: unstable-2021-01-29 -> 0.9.8

### PR DESCRIPTION
**Pull request by armijnhemel**
###### Motivation for this change

New upstream version that fixes bugs and removes the need for extra patch.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 137305` at f2707dd6 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built successfully:</summary>
  <ul>
    <li>python38Packages.telfhash</li>
    <li>python39Packages.telfhash</li>
  </ul>
</details>
<details>
  <summary>2 suggestions:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/no-python-tests.md">no-python-tests</a>

Test runner could not discover any test cases: ‘Ran 0 tests in 0.073s’
Near pkgs/development/python-modules/telfhash/default.nix:47:0:
```
   |
47 |     description = "Symbol hash for ELF files";
   | ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/no-python-tests.md">no-python-tests</a>

Test runner could not discover any test cases: ‘Ran 0 tests in 0.077s’
Near pkgs/development/python-modules/telfhash/default.nix:47:0:
```
   |
47 |     description = "Symbol hash for ELF files";
   | ^
```
</li>
  </ul>
</details>

File: pkgs/development/python-modules/telfhash/default.nix [hunk 1]
```diff
@@ -2,32 +2,33 @@
...
+    rev = "0aa7df079c00fe00e3a1147adb52f010cfa4431d";
+    sha256 = "124zajv43wx9l8rvdvmzcnbh0xpzmbn253pznpbjwvygfx16gq02";
   };
 
-  patches = [ ./telfhash-new-tlsh-hash.patch ];
-
   # The tlsh library's name is just "tlsh"
   postPatch = ''
     substituteInPlace requirements.txt --replace "python-tlsh" "tlsh"
+    substituteInPlace requirements.txt --replace "py-tlsh" "tlsh"
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/telfhash/default.nix:24]:
```suggestion
    substituteInPlace requirements.txt \
    	--replace "python-tlsh" "tlsh" \
       --replace "py-tlsh" "tlsh"
```

File: pkgs/development/python-modules/telfhash/default.nix [hunk 2]
```diff
@@ -2,32 +2,33 @@
...
   pname = "telfhash";
-  version = "unstable-2021-01-29";
+  version = "0.9.8";
 
   src = fetchFromGitHub {
     owner = "trendmicro";
     repo = "telfhash";
-    rev = "b5e398e59dc25a56a28861751c1fccc74ef71617";
-    sha256 = "jNu6qm8Q/UyJVaCqwFOPX02xAR5DwvCK3PaH6Fvmakk=";
+    rev = "0aa7df079c00fe00e3a1147adb52f010cfa4431d";
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/telfhash/default.nix:17]:
```suggestion
    rev = version;
```

File: pkgs/development/python-modules/telfhash/default.nix [hunk 3]
```diff
@@ -2,32 +2,33 @@
 , buildPythonPackage
 , fetchFromGitHub
 , capstone
+, packaging
 , pyelftools
 , tlsh
 , nose
 }:
 buildPythonPackage {
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/telfhash/default.nix:10]:
```suggestion
buildPythonPackage rec {
```

File: pkgs/development/python-modules/telfhash/default.nix [hunk 4]
```diff
@@ -2,32 +2,34 @@
...
   pname = "telfhash";
-  version = "unstable-2021-01-29";
+  version = "0.9.8";
 
   src = fetchFromGitHub {
     owner = "trendmicro";
     repo = "telfhash";
-    rev = "b5e398e59dc25a56a28861751c1fccc74ef71617";
-    sha256 = "jNu6qm8Q/UyJVaCqwFOPX02xAR5DwvCK3PaH6Fvmakk=";
+    rev = version;
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/telfhash/default.nix:17]:
```suggestion
    rev = "$v{version}";
```

File: pkgs/development/python-modules/telfhash/default.nix [hunk 5]
```diff
@@ -2,32 +2,34 @@
...
   pname = "telfhash";
-  version = "unstable-2021-01-29";
+  version = "0.9.8";
 
   src = fetchFromGitHub {
     owner = "trendmicro";
     repo = "telfhash";
-    rev = "b5e398e59dc25a56a28861751c1fccc74ef71617";
-    sha256 = "jNu6qm8Q/UyJVaCqwFOPX02xAR5DwvCK3PaH6Fvmakk=";
+    rev = "$v{version}";
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/telfhash/default.nix:17]:
```suggestion
    rev = "v${version}";
```

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).

Result of `nixpkgs-review pr 137305` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
2 packages built:
</summary>
<ul>
<li>python38Packages.telfhash</li>
<li>python39Packages.telfhash</li>
</ul>
</details>

**Pull request merged**